utils/docker/Dockerfile.el.8

```diff
@@ -177,24 +133,30 @@ ARG DAOS_TARGET_TYPE=release
 # src hasn't changed then this won't do anything, but if it has then we want to
 # ensure that latest dependencies are used.
 USER root:root
-RUN [ "$DAOS_DEPS_BUILD" != "yes" ] || \
-    { dnf upgrade --exclude=spdk,spdk-devel,dpdk-devel,dpdk,mercury-devel,mercury && \
-    dnf clean all; }
+RUN [ "$DAOS_DEPS_BUILD" != "yes" ] || {                                       \
+        dnf -y upgrade                                                         \
+            --exclude=spdk,spdk-devel,dpdk-devel,dpdk,mercury-devel,mercury && \
+        dnf -y clean all;                                                         \
+    }
 USER daos_server:daos_server
 
 ARG DEPS_JOBS=1
 
-RUN [ "$DAOS_DEPS_BUILD" != "yes" ] || \
-    { scons --build-deps=yes --jobs $DEPS_JOBS PREFIX=/opt/daos \
-      TARGET_TYPE=$DAOS_TARGET_TYPE --deps-only && \
-    ([ "$DAOS_KEEP_BUILD" != "no" ] || /bin/rm -rf build *.gz); }
+RUN [ "$DAOS_DEPS_BUILD" != "yes" ] || {                            \
+        scons --build-deps=yes --jobs $DEPS_JOBS PREFIX=/opt/daos   \
+              TARGET_TYPE=$DAOS_TARGET_TYPE --deps-only &&          \
+        ([ "$DAOS_KEEP_BUILD" != "no" ] || /bin/rm -rf build *.gz); \
+    }
+USER root:root
 
+# select compiler to use
+ARG COMPILER=gcc
```

Comment:
Moving this further up doesn't make sense, in general you only define things at late as possibly before you use them.  Having this hear will result in more load on the servers, and more images for Jenkins etc to save and there's no advantage I can see for the change.

Reply:
It needed to be moved up in another Dockerfile and was done here to keep them in sync, if you, say opened them all up with `vim -d`.

Reply:
It probably needs moving down in one of them then.

Reply:
It can't because of https://github.com/daos-stack/daos/blob/a18dc1196bca2ec52753597133b5b3c758aac512/utils/docker/Dockerfile.leap.15#L129-L146

Arguably, the above should be added to all other Dockerfiles so that any platform has the option of using the Intel Compiler.